cpp/src/core/sroptcryst.h

```diff
@@ -1127,6 +1132,34 @@ class srTOptCryst : public srTGenOptElem {
 			*(EPtrs.pEzRe) = (float)(Ety.real());
 			*(EPtrs.pEzIm) = (float)(Ety.imag());
 		}
+        else if(m_uc == 3) //Laue Reflection LD
+		{
+			complex<double> Ehx = sgH[0]*EHSPCs + piH[0]*EHSPCp;
+			complex<double> Ehy = sgH[1]*EHSPCs + piH[1]*EHSPCp;
+			complex<double> Ehz = sgH[2]*EHSPCs + piH[2]*EHSPCp; //Longitudinal component not used (?)
+		
+			//Transverse components of the output electric field in the frame of the output beam:
+			*(EPtrs.pExRe) = (float)(asymFact*Ehx.real());
+			*(EPtrs.pExIm) = (float)(asymFact*Ehx.imag());
+			*(EPtrs.pEzRe) = (float)(asymFact*Ehy.real());
+			*(EPtrs.pEzIm) = (float)(asymFact*Ehy.imag());
+		}
```

Comment:
This seems to be exactly the same code as for "if(m_uc == 1)" (?)
Merges it there as: "if((m_uc == 1) || (m_uc == 3))".